mobile/lib/providers/asset.provider.dart

```diff
@@ -78,27 +80,48 @@ class AssetNotifier extends StateNotifier<bool> {
   }
 
   Future<bool> deleteLocalOnlyAssets(
+    BuildContext context,
     Iterable<Asset> deleteAssets, {
     bool onlyBackedUp = false,
   }) async {
     _deleteInProgress = true;
     state = true;
     try {
+      // Filter local assets based on the backup status (`isRemote`).
+      // If `onlyBackedUp` is true, only select assets that are also present on the server (`isRemote`).
+      final assetsToDelete = deleteAssets
+          .where((e) => e.isLocal && (!onlyBackedUp || e.isRemote))
+          .toList();
+
+      if (assetsToDelete.isEmpty) {
+        // No assets to delete, exit early
+        ImmichToast.show(
+          context: context,
+          msg: 'No backed-up assets selected for deletion.',
+          gravity: ToastGravity.BOTTOM,
+        );
+        return false;
+      }
+
+      // Continue with deletion of the filtered assets
       final assets = onlyBackedUp
-          ? deleteAssets.where((e) => e.storage == AssetState.merged)
-          : deleteAssets;
```

Comment:
This section of code already filters the total list passed based on the backed up status of the asset. i.e, `e.storage == AssetState.merged` essentially filters only those that are already backed up. You added the same filtering more explicitly before this thus making this iteration useless